test/typedarray/delete.js

```diff
@@ -0,0 +1,36 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+if (this.WScript && this.WScript.LoadScriptFile) { // Check for running in ch
+    this.WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+    this.WScript.LoadScriptFile("util.js");
+}
+
+var tests = [
+    {
+        name: "Typed arrays support delete of non-indexed properties",
+        body: function () {
+            const ta = Int8Array.of(42);
+
+            ta.non_indexed = 'whatever';
+            assert.areEqual('whatever', ta.non_indexed, "ta.non_indexed is set to 'whatever'");
+
+            delete ta.non_indexed;
+            assert.areEqual(undefined, ta.non_indexed, "ta.non_indexed has been deleted");
+        }
+    },
+    {
+        name: "Typed arrays don't support delete of indexed properties",

```

Comment:
Might be worth adding another case for attempting to delete an indexed property in non-strict mode, and ensuring that the property is not deleted and that it does not throw?

Reply:
Line 29 tests silent failure in non-strict mode and line 31 tests exception thrown in strict mode :)

Reply:
Oh of course, sorry for missing that!